jwst/outlier_detection/outlier_detection.py

```diff
@@ -403,96 +400,44 @@ def flag_cr(sci_image, blot_image, **pars):
-    tmp2 = np.zeros(tmp1.shape, dtype=np.int32)
-    ndimage.convolve(tmp1, kernel, output=tmp2, mode='nearest', cval=0)
-    #
-    #
-    #    COMPUTATION PART II
-    #
-    #
-    # Create a second CR Mask
-    xt2 = scl2 * blot_deriv + snr2 * ta
-
-    np.logical_not(np.greater(diff_noise, xt2) & np.less(tmp2, 9), cr_mask)
-
-    #
-    #
-    #    COMPUTATION PART III
-    #
-    #
-    # Flag additional cte 'radial' and 'tail' pixels surrounding CR
-    # pixels as CRs
-
-    # In both the 'radial' and 'length' kernels below, 0=good and
-    # 1=bad, so that upon convolving the kernels with cr_mask, the
-    # convolution output will have low->bad and high->good from which
-    # 2 new arrays are created having 0->bad and 1->good. These 2 new
-    # arrays are then AND'ed to create a new cr_mask.
-
-    # recast cr_mask to int for manipulations below; will recast to
-    # Bool at end
-    cr_mask_orig_bool = cr_mask.copy()
-    cr_mask = cr_mask_orig_bool.astype(np.uint8)
-
-    # make radial convolution kernel and convolve it with original cr_mask
-    cr_grow_kernel = np.ones((grow, grow))
-    cr_grow_kernel_conv = cr_mask.copy()
-    ndimage.convolve(cr_mask, cr_grow_kernel, output=cr_grow_kernel_conv)
-
-    # make tail convolution kernel and (shortly) convolve it with
-    # original cr_mask
-    cr_ctegrow_kernel = np.zeros((2 * ctegrow + 1, 2 * ctegrow + 1))
-    cr_ctegrow_kernel_conv = cr_mask.copy()
-
-    # which pixels are masked by tail kernel depends on readout direction
-    # We could put useful info in here for CTE masking if needed.  Code
-    # remains below.  For now, we set to zero, which turns off CTE masking.
-    ctedir = 0
-    if (ctedir == 1):
-        cr_ctegrow_kernel[0:ctegrow, ctegrow] = 1
-    if (ctedir == -1):
-        cr_ctegrow_kernel[ctegrow + 1:2 * ctegrow + 1, ctegrow] = 1
-    if (ctedir == 0):
-        pass
-
-    # finally do the tail convolution
-    ndimage.convolve(cr_mask, cr_ctegrow_kernel, output=cr_ctegrow_kernel_conv)
-
-    # select high pixels from both convolution outputs; then 'and' them to
-    # create new cr_mask
-    where_cr_grow_kernel_conv = np.where(cr_grow_kernel_conv < grow * grow,
-                                         0, 1)
-    where_cr_ctegrow_kernel_conv = np.where(cr_ctegrow_kernel_conv < ctegrow,
-                                            0, 1)
-
-    # combine masks and cast back to Bool
-    np.logical_and(where_cr_ctegrow_kernel_conv,
-                   where_cr_grow_kernel_conv, cr_mask)
-    cr_mask = cr_mask.astype(bool)
+    # create the outlier mask
+    if pars.get('resample_data', True):  # dithered outlier detection
+        blot_data += subtracted_background
+        diff_noise = np.abs(sci_data - blot_data)
+
+        # create an initial mask based on
+        # a scaled version of the derivative image (dealing with interpolating issues?)
+        # and the standard n*sigma above the noise
+        t2 = scl1 * blot_deriv + snr1 * err_data
```

Comment:
The blot_data contribution is to add photon noise to the threshold.  But this is (now) included in the err_data array that includes read, photon, and flat field noise.  Removing it so that the photon noise is not added twice.